mlflow/data/delta_dataset_source.py

```diff
@@ -86,6 +110,45 @@ def _is_databricks_uc_table(self):
                 and catalog_name != DATABRICKS_SAMPLES_CATALOG_NAME
             )
 
+    def _lookup_table_id(self, table_name):
+        try:
+            req_body = message_to_json(GetTable(full_name_arg=table_name))
+            _METHOD_TO_INFO = extract_api_info_for_service(
+                UnityCatalogService, _REST_API_PATH_PREFIX
+            )
+            db_creds = get_databricks_host_creds()
+            endpoint, method = _METHOD_TO_INFO[GetTable]
+            # We need to replace the full_name_arg in the endpoint definition with
+            # the actual table name for the REST API to work.
+            final_endpoint = endpoint.replace("{full_name_arg}", table_name)
+            resp = call_endpoint(
+                host_creds=db_creds,
+                endpoint=final_endpoint,
+                method=method,
+                json_body=req_body,
+                response_proto=GetTableResponse,
+            )
+            return resp.table_id
+        except Exception:
+            return None
+
+    def _resolve_table_name(self, table_name) -> Optional[str]:
```

Comment:
Can we reuse/share code with UC model name resolution for htis?

Reply:
Same as above.